Atomic/rpmwriter.py

```diff
@@ -0,0 +1,369 @@
+# http://ftp.rpm.org/max-rpm/s1-rpm-file-format-rpm-file-format.html
+# The same documentation was used to initially write this module.
+#
+# Only tags that are actually used in the code were added to the RpmWriter class.
+# The complete list is available here:
+# https://github.com/rpm-software-management/rpm/blob/master/lib/rpmtag.h
+
+class RpmWriter(object):
+
+    MAGIC = [0xed, 0xab, 0xee, 0xdb]
+    MAJOR = [0x3]
+    MINOR = [0x0]
+    BINARY_TYPE = [0x0, 0x0]
+    NOARCH = [0xFF, 0xFF]
+    OS = [0x0, 0x1]
+    SIGNATURE_TYPE = [0x0, 0x5]
+    RESERVED = [0x0] * 16
+
+    HEADER_MAGIC = [0x8e, 0xad, 0xe8]
+    HEADER_VERSION = [0x01]
+    HEADER_RESERVED = [0x0] * 4
+
+    SIGNATURE_MAGIC = [0x8e, 0xad, 0xe8]
+    SIGNATURE_VERSION = [0x01]
+    SIGNATURE_RESERVED = [0x0] * 4
+
+    RPMSIGTAG_PAYLOADSIZE = 1007
+
+    RPMTAG_NAME = 1000
+    RPMTAG_VERSION = 1001
+    RPMTAG_RELEASE = 1002
+    RPMTAG_SUMMARY = 1004
+    RPMTAG_DESCRIPTION = 1005
+    RPMTAG_LICENSE_ = 1014
+    RPMTAG_GROUP = 1016
+    RPMTAG_CHANGELOG = 1017
+    RPMTAG_URL = 1020
+    RPMTAG_OS = 1021
+    RPMTAG_ARCH = 1022
+    RPMTAG_FILESIZES = 1028
+    RPMTAG_DIRINDEXES = 1116
+    RPMTAG_BASENAMES = 1117
+    RPMTAG_DIRNAMES = 1118
+    RPMTAG_FILEMODES = 1030
+    RPMTAG_FILERDEVS = 1033
+    RPMTAG_FILEMTIMES = 1034
+    RPMTAG_FILEDIGESTS = 1035
+    RPMTAG_FILELINKTOS = 1036
+    RPMTAG_FILEFLAGS = 1037
+    RPMTAG_FILEUSERNAME = 1039
+    RPMTAG_FILEGROUPNAME = 1040
+
+    RPMTAG_PROVIDENAME = 1047
+    RPMTAG_REQUIRENAME = 1049
+    RPMTAG_REQUIREVERSION = 1050
+    RPMTAG_CONFLICTFLAGS = 1053
+    RPMTAG_CONFLICTNAME = 1054
+    RPMTAG_CONFLICTVERSION = 1055
+
+    RPMTAG_OBSOLETENAME	= 1090
+    RPMTAG_FILEINODES = 1096
+    RPMTAG_FILELANGS = 1097
+
+    RPMTAG_PAYLOADFORMAT = 1124
+    RPMTAG_PAYLOADCOMPRESSOR = 1125
+
+    RPMTAG_FILEDIGESTALGO = 5011
+
+    RPM_SPEC_FILEMODE = (1 << 8)
+    RPM_SPEC_DIRMODE = (1 << 9)
+
+    RPMFILE_CONFIG = (1 <<  0)
+
+    RPMTAG_PAYLOADDIGEST = 5092
+    RPMTAG_PAYLOADDIGESTALGO = 5093
+
+    PGPHASHALGO_SHA1 = 2
+
+    def get_sha1(self, path):
+        m = hashlib.sha1()
+        try:
+            with open(path, 'r') as f:
+                while True:
+                    data = os.read(f.fileno(), 4096)
+                    if len(data) == 0:
+                        break
+                    m.update(data)
+            return m.hexdigest()
+        except IOError:
+            return ""
+
+    def add_require(self, name, version):
+        self.require.append([name, version])
+
+    def add_provide(self, name):
+        self.provide.append(name)
+
+    def add_obsolete(self, name):
+        self.obsolete.append(name)
+
+    def add_conflict(self, name, version):
+        self.conflict.append([name, version])
+
+    def __init__(self, out, root, name, version, release, summary='', description='', license_='gpl2', changelog='', url='', group='', stderr=sys.stderr, whitelist=None):
```

Comment:
rpm complains if no license is specified, so I've just used the same license of `atomic` itself